build/multilocale.js

```diff
@@ -205,16 +205,97 @@ function L10nManager(gaiaDir,
 
       var doc = utils.getDocument(content);
 
-      buildL10nMeta(htmlFile, doc);
+      return buildL10nMeta(htmlFile, doc).then(function() {
+        if (self.localeBasedir) {
+          // Copy resource files into build_stage directory
+          getL10nResources(htmlFile, webapp, doc);
+        }
+      });
+    });
+
+    return Promise.all(localizeHtmlFiles);
+  }
 
-      if (self.localeBasedir) {
-        // Copy resource files into build_stage directory
-        getL10nResources(htmlFile, webapp, doc);
+  function hasCommand(cmdName) {
+    return utils.getEnvPath().some(function(path) {
+      try {
+        var cmd = utils.getFile(path, cmdName);
+        return cmd.exists();
+      } catch (e) {
+        // path not found
       }
+      return false;
+    });
+  }
+
+  function getLastGitCommitTimestamp(path) {
+    var gitDir = utils.getFile(path, '.git');
+    var git = new utils.Commander('git');
+    var stderr, stdout;
+    var args = [
+      '--git-dir=' + gitDir.path,
+      'log',
+      '--format=%ct',
+      'HEAD',
+      '-1'
+    ];
+
+    return new Promise(function(resolve, reject) {
+      var cmdOptions = {
+        stdout: function(data) {
+          stdout = data;
+        },
+        stderr: function(data) {
+          stderr = data;
+        },
+        done: function(data) {
+          if (data.exitCode === 0) {
+            resolve(parseInt(stdout.trim()) * 1000);
+          } else {
+            reject(stderr);
+          }
+        }
+      };
+
+      git.initPath(utils.getEnvPath());
+      git.runWithSubprocess(args, cmdOptions);
```

Comment:
omg. This is embarrassing. It actually is sync. This makes the patch so much simpler. Thanks!